freqtrade/strategy/interface.py

```diff
@@ -707,35 +717,39 @@ def _analyze_ticker_internal(self, dataframe: DataFrame, metadata: dict) -> Data
         # always run if process_only_new_candles is set to false
         if (not self.process_only_new_candles or
                 self._last_candle_seen_per_pair.get(pair, None) != dataframe.iloc[-1]['date']):
+
             # Defs that only make change on new candle data.
             dataframe = self.analyze_ticker(dataframe, metadata)
+
             self._last_candle_seen_per_pair[pair] = dataframe.iloc[-1]['date']
-            self.dp._set_cached_df(
-                pair, self.timeframe, dataframe,
-                candle_type=self.config.get('candle_type_def', CandleType.SPOT))
+
+            candle_type = self.config.get('candle_type_def', CandleType.SPOT)
+            self.dp._set_cached_df(pair, self.timeframe, dataframe, candle_type=candle_type)
+            self.dp.emit_df((pair, self.timeframe, candle_type), dataframe)
```

Comment:
Sorry if I am missing something, but is it true that adding this here means all FT bots will now try to `self.dp.emit_df`? Even if they arent trying to share dataframes via websockets?

Reply:
the leader won't take that decision (neither should this part of the bot).
if there's a subscriber - it'll receive the dataframe.

if there's no subscriber - no one will get the dataframe.